### ISSUE TITLE
Node process killed due to lack of memory

### ISSUE THREAD

**Issue by jakubgs** [sourcecred/sourcecred#2086]
I'm running SourceCred on an 4 orgs with a total of 422 repos, and when I pass the `compute-cred` stage it dies:
```
 GO   compute-cred
DONE  compute-cred: 1h 9m
/opt/sourcecred/gen.sh: line 41: 2933566 Killed node ${SOURCECRED_SCRIPT} load --weights "${SOURCECRED_PATH}/weights.json" --project "${SOURCECRED_PATH}/all-orgs.json"
```
I'm running `0.6.0` and this happens on a host with 8 GB of RAM and 8 GB of SWAP.

Here's the `dmesg` output: https://gist.github.com/jakubgs/b8a12bc6659a2241b8bbccbf90540461

Is there anything I can do to fix this? And kind of flags I could pass?

**Comment by decentralion** [comment]
Currently cred computation runs TimelineCred, which produces an intermediate data structure of size `(N+E) * T`, where `N` is number of nodes, `E` is number of edges, and `T` is number of intervals. I would guess that this is the bottleneck. Once we switch to CredRank (https://github.com/sourcecred/sourcecred/issues/1686) the data will drop in size to approximately just `N+E`, i.e. proportional to the Graph size, so that will likely fix this. I'd estimate we're likely to make that swap sometime in the next two months, so I would advise holding tight.

For my own edification -- would you report the size of the GitHub graph you generated? `ls -lah output/graphs/sourcecred/github/graph.json` from within your instance directory.

**Comment by jakubgs** [comment]
I don't have such a file because the generation fails.

**Comment by decentralion** [comment]
The file I mentioned is generated by the `sourcecred graph` command, which is run before `sourcecred score`. So it is actually being generated as part of the workflow. But I see that there may not be an easy way to get that data given that the action fails.

**Comment by jakubgs** [comment]
Oh, I don't actually run any of these commands(`graph` or `score`). What I run is:
```sh
# This generates the project JSON file including all repos
echo " * Creating an 'all-orgs' project..."
node ${SOURCECRED_SCRIPT} gen-project all-orgs \
  $(printf -- "--github @%s " "${SOURCECRED_ORG_NAMES[@]}") \
  > ${SOURCECRED_PATH}/all-orgs.json

# Collect the actual data from all repos
echo " * Loading repo data for all organizations"
node ${SOURCECRED_SCRIPT} load \
  --weights "${SOURCECRED_PATH}/weights.json" \
  --project "${SOURCECRED_PATH}/all-orgs.json"
```
Because I'm combining multiple repos from multiple orgs into one output.

**Comment by decentralion** [comment]
Oh, I see. You're actually using the legacy alpha interface, which is deprecated and totally unsupported. We're in the process of migrating all users to the new instance system; you can see the [template instance](https://github.com/sourcecred/template-instance) and our [own official Cred instance](https://github.com/sourcecred/cred). The README on the template instance has the best info right now on how to use SourceCred.

The major upsides of the instance system are:
- all config is stored as data and checked-in to the master branch
- all state (including cache) is stored within the SourceCred directory, improving reproducibility and debuggability
- we have GitHub actions for automatically updating the Cred and deploying to GitHub pages
- Grain distributions are tracked by the instance and can be automated

Overall you will find it to be a huge improvement over the CLI you've been using. We're still in the process of polishing the docs and rough edges (official release by end of September).

**Comment by jakubgs** [comment]
I see. I guess I have yet another thing to update. Does this new setup support combining multiple orgs though?

**Comment by jakubgs** [comment]
Based of the format of this file it seems like it would support multiple repos:
https://github.com/sourcecred/cred/blob/master/config/plugins/sourcecred/github/config.json
Though I would have to automate somehow adding new repos to this.

**Comment by decentralion** [comment]
> Also, I assume that that template is intended for use with 0.6.0, correct?

No. The template is intended for use with 0.7.0, which isn't officially released yet, but we have a number of pre-releases that have been published to npm.

0.6.0 was an alpha / preview / prototype, and never actually supported for external use. When we officially cut 0.7.0, it'll be our first supported release.

> Though I would have to automate somehow adding new repos to this.

Yeah. We will want to add org-level support to the config. It wouldn't be hard, we already have all the pieces, it's just a matter of wiring it together. If you want to hack on this I can give some guidance.
